### PR TITLE
Upgrade debian based systems

### PR DESCRIPTION
**Pull request by Mr-Pi**
* task to upgrade and show upgradable packages on debian systems
* Add Doku
* Add LICENSE
* Updated documentation
* Restart services, when needed after upgrade(can be disabled by `upgrade_restart_services` option)

### CODE REVIEW

File: tasks/upgrade_debian.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+- name: Update packages list
+  apt: update_cache=yes
```
Comment (SlothOfAnarchy) [inline tasks/upgrade_debian.yml]: please indent

File: tasks/upgrade_debian.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  changed_when: false
+
+- name: List packages to upgrade (1/2)
+  shell: aptitude -q -F%p --disable-columns search '~U'
```
Comment (SlothOfAnarchy) [inline tasks/upgrade_debian.yml]: is aptitude available on all Debian-based releases?

File: tasks/upgrade_debian.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  register: updates
+  failed_when: updates.rc != 0 and updates.rc != 1
+  changed_when: False
+
+- name: List packages to upgrade (2/2)
+  debug: msg="{{ updates.stdout_lines | count }} packages to upgrade ({{ updates.stdout_lines | join(', ') }})"
+  when: updates.stdout_lines | count > 1
+
+- name: Upgrade packages
+  apt: upgrade=safe
```
Comment (SlothOfAnarchy) [inline tasks/upgrade_debian.yml]: indent

File: tasks/upgrade_debian.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- name: Notify distribution version upgrade
+  debug: msg="Debian has been upgraded from {{ ansible_lsb.release }} to {{ new_release.stdout }}"
+  when: ansible_lsb.release != new_release.stdout
+
+- name: List services to restart (1/2)
+  shell: checkrestart | grep ^service | awk '{print $2}'
+  register: services
+  changed_when: false
+
+- name: List services to restart (2/2)
```
Comment (SlothOfAnarchy) [inline tasks/upgrade_debian.yml]: Would it be a good idea to include a task that restarts these services?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Role Name
```
Comment (SlothOfAnarchy) [inline README.md]: change the name

File: README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+A debian based distribution at all hosts to be upgraded
+
+## Role Variables
+
+
+```yml
+```
+
+
+## Dependencies
```
Comment (SlothOfAnarchy) [inline README.md]: remove all these unnecessary headings

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

**Comment by Mr-Pi** [comment]
Please rereview

**Comment by SlothOfAnarchy** [comment]
@Mr-Pi If you set the role name in readme and remove the empty sections, you're good to go!

**Comment by Mr-Pi** [comment]
@SlothOfAnarchy what you mean, they are set correctly

**Comment by SlothOfAnarchy** [comment]
I'm sorry, I didn't see the changed role name. `Role Variables`, `Dependencies` and `Example Playbook` are still empty sections though. Please fill or remove them.

File: README.md [hunk 7]
```diff
@@ -11,9 +11,9 @@ A debian based distribution at all hosts to be upgraded and the following packag
 
 ## Role Variables
 
-## Dependencies
-
-## Example Playbook
+| Option                   | Type    | Default | Description                                                                      | Required |
+|--------------------------|---------|---------|----------------------------------------------------------------------------------|:--------:|
+| upgrade_restart_services | boolean | true    | Automatically restart services, when needed(for ex. linked libaries are updated) |     N    |
```
Comment (SlothOfAnarchy) [inline README.md]: use code formatting

File: tasks/upgrade_debian.yml [hunk 8]
```diff
@@ -40,4 +40,4 @@
     name: "{{ item }}"
     state: restarted
   with_items: "{{ services.stdout_lines }}"
-  when: services.stdout_lines
+  when: services.stdout_lines and upgrade_restart_services
```
Comment (SlothOfAnarchy) [inline tasks/upgrade_debian.yml]: we could also use a tag for this, what do you think?

File: README.md [hunk 9]
```diff
@@ -1,26 +1,20 @@
...
+# upgrade
 
 Simple ansible playbook to upgrade servers, based on [debian_upgrade.yml](https://gist.github.com/maethor/380676f6b1cec8cc7439) by Guillaume Subiron (maethor)
 
 
 ## Requirements
 
-A debian based distribution at all hosts to be upgraded
+A debian based distribution at all hosts to be upgraded and the following packages:
+	* aptitude
```
Comment (SlothOfAnarchy) [inline README.md]: fix list formatting

**Review by SlothOfAnarchy — DISMISSED** [review]

**Comment by Mr-Pi** [comment]
Is fixed

File: README.md (hunk 7 above)
Comment (Mr-Pi) [reply README.md]: what you mean?

**Comment by Mr-Pi** [comment]
Okay, please rereview :)

File: README.md (hunk 7 above)
Comment (SlothOfAnarchy) [inline README.md]: `upgrade_restart_services` instead of upgrade_restart_services

**Review by SlothOfAnarchy — APPROVED** [review]
Yes, now I like it, thanks! We can still talk about using a tag instead of the var `upgrade_restart_services`.

**Pull request merged**